react/features/rtcstats/RTCStats.ts

```diff
@@ -47,9 +47,44 @@ function connectionFilter(config: any) {
  */
 class RTCStats {
     trace: any;
-    initialized = false;
+    options?: InitOptions;
+    isPeerConnectionWrapped = false;
     connStateEvents: any = [];
 
+    /**
+     * Initialize the rtcstats components, if the options have changed.
+     *
+     * @param {Oject} newOptions -.
+     * @returns {void}
+     */
+    maybeInit(newOptions: InitOptions) {
+        const oldOptions = this.options;
+        const changed = !oldOptions || (oldOptions.endpoint !== newOptions.endpoint
+            || oldOptions.meetingFqn !== newOptions.meetingFqn
+            || oldOptions.pollInterval !== newOptions.pollInterval
+            || oldOptions.sendSdp !== newOptions.sendSdp
+            || oldOptions.useLegacy !== newOptions.useLegacy);
+
+        if (changed) {
+            this.reset();
+
+            if (newOptions.meetingFqn && newOptions.endpoint) {
+                this.init(newOptions);
+            }
+        }
+    }
+
+    /**
+     * Wrapper method for the underlying trace object to be used as a static reference from inside the wrapped
+     * PeerConnection.
+     *
+     * @param {any[]} data - the stats entry to send to the rtcstats endpoint.
+     * @returns {void}
+     */
+    statsEntry(...data) {
+        this.trace?.statsEntry(data);
+    }
```

Comment:
I'm not 100% certain, but I don't think this solves the problem that `trace()` gets passed on to the rtcstats node module, which will hold on to that no matter what you do with the local reference to `trace()`.

The proposal in my previous comment would involve modifying the actual rtcstats node module. And add a setter there, which you could call from here if `maybeInit()` decides to reinitialize things. Which is obviously way more complicated. But I'm not sure I see an easier solution to the rtcstats node module continuing to use an old `trace()` reference.

Reply:
Unfortunately, the rtcstats module in its current state exports a JavaScript function and so it can't have any properties or methods nor can we modify its state.